src/TSID/src/QPFixedBaseTSID.cpp

```diff
@@ -0,0 +1,619 @@
+        cost.get().tmp.noalias() = A.transpose() * cost.get().weight.asDiagonal();
+        m_pimpl->hessian.noalias() += cost.get().tmp * A;
+        m_pimpl->gradient.noalias() -= cost.get().tmp * b;
+    }
+
+    // compute the constraints
+    std::size_t index = 0;
+    for (const auto& constraint : m_pimpl->constraints)
+    {
+        // check if the number of constraints changed
+        if (m_pimpl->numberOfConstraints < index + constraint.get().task->size())
+        {
+            log()->error("{} The number of constraints changed. Please call finalize() again. "
+                         "Expected number of constraints: {}. Adding the constrained named {}, the "
+                         "number of constraints becomes {}.",
+                         logPrefix,
+                         m_pimpl->numberOfConstraints,
+                         constraint.get().task->getDescription(),
+                         index + constraint.get().task->size());
+            return false;
+        }
+
+        Eigen::Ref<const Eigen::MatrixXd> A = constraint.get().task->getA();
+        Eigen::Ref<const Eigen::VectorXd> b = constraint.get().task->getB();
+
+        m_pimpl->constraintMatrix.middleRows(index, constraint.get().task->size()) = A;
+        m_pimpl->upperBound.segment(index, constraint.get().task->size()) = b;
+
+        if (constraint.get().task->type() == System::LinearTask::Type::inequality)
+        {
+            m_pimpl->lowerBound.segment(index, constraint.get().task->size())
+                .setConstant(-OsqpEigen::INFTY);
+        } else
+        {
+            assert(constraint.get().task->type() == System::LinearTask::Type::equality);
+
+            m_pimpl->lowerBound.segment(index, constraint.get().task->size()) = b;
+        }
+
+        index += constraint.get().task->size();
+    }
+
+    // update the solver
+    if (!m_pimpl->isFirstIteration)
+    {
+        if (!m_pimpl->updateSolver())
+        {
+            log()->error("{} Unable to update the QP solver.", logPrefix);
+            return false;
+        }
+    } else
+    {
+        if (!m_pimpl->initializeSolver())
+        {
+            log()->error("{} Unable to run the QP solver the first time.", logPrefix);
+            return false;
+        }
+        m_pimpl->isFirstIteration = false;
+    }
+
+    // solve the QP
+    if (!m_pimpl->solver.solve())
+    {
+        log()->error("{} Unable to to solve the problem.", logPrefix);
+        return false;
+    }
+
+    // retrieve the solution
+    constexpr std::size_t spatialAccelerationSize = 6;
+    const std::size_t joints = m_pimpl->robotAccelerationVariable.size - spatialAccelerationSize;
+
+    m_pimpl->solution.jointTorques
+        = m_pimpl->solver.getSolution().segment(m_pimpl->robotAccelerationVariable.offset
+                                                    + m_pimpl->robotAccelerationVariable.size,
+                                                joints);
```

Comment:
`Solution` contains also the robot acceleration. I think you should fill also that